yunchipang/goodhome-frontend
# feature/chat

- [x] implement real time chat using websocket
- [x] display history chat in the chatroom, stored history chat messages in db
- [x] prettify react frontend, display messages on left and right side of the chatroom

test instructions
1. run your django backend on localhost port 8000 AND have a redis server running on port 6379 (see instructions [here](https://github.com/yunchipang/goodhome/pull/15))
1. log into app via `/signuplogin`
2. visit buyer-portal and input a zipcode of your choice
<img width="1470" alt="Screenshot 2024-04-09 at 7 39 43 PM" src="https://github.com/yunchipang/goodhome-frontend/assets/48413155/72b3ce45-1a3d-4c49-9ee2-c3834af98e5f">
3. find the seller you'd want to chat with, click on the "Chat" button
<img width="1470" alt="Screenshot 2024-04-09 at 7 39 52 PM" src="https://github.com/yunchipang/goodhome-frontend/assets/48413155/2f460022-84ad-448b-8989-1430e4e53bc8">
4. enters the chat room with the seller, start chatting!
<img width="1470" alt="Screenshot 2024-04-09 at 7 40 05 PM" src="https://github.com/yunchipang/goodhome-frontend/assets/48413155/bfe63446-a37e-47a2-b12b-6c8165bee5bd">
5. visit `/chat` to view the list of users you've had conversation with
<img width="1470" alt="Screenshot 2024-04-09 at 7 40 16 PM" src="https://github.com/yunchipang/goodhome-frontend/assets/48413155/736d85dd-3287-400f-a0d4-db670f292cd4">
6. click on the "Chat" button to enter the chatroom with chat history retrieved
<img width="1470" alt="Screenshot 2024-04-09 at 7 40 25 PM" src="https://github.com/yunchipang/goodhome-frontend/assets/48413155/633ee53f-b0db-46e1-b3a6-6941f9e24c6a">

If you'd like to test the real time chat feature on your local machine, run the frontend code in 2 separate terminals, and log in with 2 different users.
```
// in terminal 1
$ PORT=3000 npm start
// in terminal 2
$ PORT=3001 npm start
```
